bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/api/EvccAPI.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.evcc.internal.api;
+
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.EVCC_REST_API;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.LONG_CONNECTION_TIMEOUT_MILLISEC;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.evcc.internal.api.dto.Result;
+import org.openhab.binding.evcc.internal.api.dto.Status;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link EvccAPI} is responsible for API calls to evcc.
+ * 
+ * @author Florian Hotze - Initial contribution
+ */
+@NonNullByDefault
+public class EvccAPI {
+    private final Logger logger = LoggerFactory.getLogger(EvccAPI.class);
+    private final Gson gson = new Gson();
+    private String host = "";
+
+    public EvccAPI(String host) {
+        this.host = host;
+    }
+
+    /**
+     * Make a HTTP request.
+     * 
+     * @param url full request URL
+     * @param method reguest method, e.g. GET, POST
+     * @return the response body
+     * @throws IOException if HTTP request failed
+     */
+    private @Nullable String httpRequest(String url, String method) throws EvccApiException {
+        try {
+            String response = HttpUtil.executeUrl(method, url, LONG_CONNECTION_TIMEOUT_MILLISEC);
+            logger.trace("{} {} - {}", method, url, response);
+            return response;
+        } catch (IOException e) {
+            throw new EvccApiException("HTTP request failed for URL " + url, e);
+        }
+    }
+
+    // End utility functions
+
+    // API calls to evcc
+    /**
+     * Get the status from evcc.
+     * 
+     * @param host hostname of IP address of the evcc instance
+     * @return {@link org.openhab.binding.evcc.internal.api.dto.Result} object or null
+     * @throws IOException if HTTP request failed
+     * @throws JsonSyntaxException if response JSON read failed
+     */
+    public @Nullable Result getResult() throws EvccApiException {
+        final String response = httpRequest(this.host + EVCC_REST_API + "state", "GET");
+        try {
+            Status status = gson.fromJson(response, Status.class);
+            if (status == null) {
+                return null;
+            }
```

Comment:
This is inconsistent as caller will need to handle errors in two different ways: Checking for null and catching exception. If you change this to something like:
```suggestion
            if (status == null) {
                throw new EvccApiException("Status is null");
            }
```

you can drop `@Nullable` from this method as well as all set methods below.

Reply:
See 7ade98f